tests/test_freqtradebot.py

```diff
@@ -3545,32 +3545,29 @@ def test_get_real_amount_no_trade(default_conf, buy_order_fee, caplog, mocker, f
                    caplog)
 
 
-def test_get_real_amount_stake(default_conf, trades_for_order, buy_order_fee, fee, mocker):
-    trades_for_order[0]['fee']['currency'] = 'ETH'
-
-    mocker.patch('freqtrade.exchange.Exchange.get_trades_for_order', return_value=trades_for_order)
-    amount = sum(x['amount'] for x in trades_for_order)
-    trade = Trade(
-        pair='LTC/ETH',
-        amount=amount,
-        exchange='binance',
-        fee_open=fee.return_value,
-        fee_close=fee.return_value,
-        open_rate=0.245441,
-        open_order_id="123456"
-    )
-    freqtrade = get_patched_freqtradebot(mocker, default_conf)
-
-    # Amount does not change
-    assert freqtrade.get_real_amount(trade, buy_order_fee) == amount
-
-
-def test_get_real_amount_no_currency_in_fee(default_conf, trades_for_order, buy_order_fee,
-                                            fee, mocker):
+@pytest.mark.parametrize(
+    'fee_cost, fee_currency, fee_reduction_amount, use_ticker_rate, expected_log', [
+        (None, 'ETH', 0, True, None),
+        (0.004, None, 0, True, None),
+        (0.00094518, "BNB", 0, True, (
+            'Fee for Trade Trade(id=None, pair=LTC/ETH, amount=8.00000000, open_rate=0.24544100,'
+            ' open_since=closed) [buy]: 0.00094518 BNB - rate: None'
+        )),
+        (0.004, "LTC", 0.004, False, (
+            'Applying fee on amount for Trade(id=None, pair=LTC/ETH, amount=8.00000000, '
+            'open_rate=0.24544100, open_since=closed) (from 8.0 to 7.996).'
```

Comment:
can you remove the text here also

Reply:
Should I not assert it at all? Most of the previous test cases weren't asserting a log line. And the log line on line 3554 is in a different format, the `Fee for Trade` one. Simply passing in a value won't work for that case and then this would be the only version that even asserts. Does that make sense?

Reply:
I'd leave it as is to be honest.

piecing the logstring together might work - but will make changes to it VERY difficult for future editors.

Reply:
One additional request though - could you add comments to each parameter-line documenting what we're trying to test? 
Currently it's pretty difficult to determine what exactly is trying to be tested...